app/javascript/src/components/editor/Editor.svelte

```diff
@@ -146,6 +153,22 @@
     })
   }
 
+  function isInherentlyHidden(item) {
+    if (item.hidden) {
+      return true
+    }
+
+    if (item.parent) {
+      const parent = $sortedItems.find((parentItem) => parentItem.id === item.parent)
+
+      if (parent) {
+        return isInherentlyHidden(parent)
+      }
+    }
+
+    return false
+  }
```

Comment:
I think we can do with a few fewer indents

```suggestion
  function isInherentlyHidden(item) {
    if (item.hidden) return true
    if (!item.parent) return false
    
    const parent = $sortedItems.find((parentItem) => parentItem.id === item.parent)
    if (!parent) return false
    
    return isInherentlyHidden(parent)
  }
```